File: mymixtape-api/models/request.go
package models

type AccessTokenRequest struct {
	Code	string	`json:"code"`
}

type CombinePlaylistRequest struct {
	UserID		string		`form:"userId"`
	Name		string		`form:"name"`
	Description	string		`form:"description"`
	PlaylistIDs	[]string	`form:"playlistIds"`
}

type GetCurrentUsersPlaylistsRequestQueryParameters struct {
	Limit	string	`form:"limit"`
	Offset	string	`form:"offset"`
}

File: mymixtape-app/src/types/api/request.ts
type ApiRequest = {
  token?: string;
};

export interface GetUserProfile extends ApiRequest {}

export interface CombinePlaylistRequest extends ApiRequest {
  user_id: string;
  name: string;
  description: string;
  playlist_ids: string[];
}

export interface GetAccessTokenRequest extends ApiRequest {
  code: string;
}

export interface GetUserPlaylistsRequest extends ApiRequest {
  offset: number;
  limit: number;
}
